src/main/java/seedu/weme/ui/ViewPanel.java

```diff
@@ -33,15 +33,21 @@
 
     public ViewPanel(ObservableValue<Meme> observableMeme) {
         super(FXML);
+        cardPane.heightProperty().addListener((observable, oldValue, newValue) -> {
+            // Force ImageView to a size based on the window's height. 75 seems to restrict it.
+            double newHeight = (double) newValue - 75 - description.getHeight() - tags.getHeight();
+            display.setFitHeight(newHeight);
+        });
+        display.fitWidthProperty().bind(cardPane.widthProperty());
         observableMeme.addListener((observable, oldValue, newValue) -> {
-            display.setImage(new Image(newValue.getImagePath().toUrl().toString(),
-                    400, 350,
-                    true, true, true));
             description.setText(newValue.getDescription().value);
             tags.getChildren().clear();
             newValue.getTags().stream()
                     .sorted(Comparator.comparing(tag -> tag.tagName))
                     .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+            display.setImage(new Image(newValue.getImagePath().toUrl().toString(),
+                    display.getFitWidth(), display.getFitHeight(),
```

Comment:
Should probably load the image in full resolution. Otherwise, if the use enters `view` at a low resolution and then increases the window size, the image might look blur.